src/mattress/TatamiNumericPointer.py

```diff
@@ -454,3 +455,31 @@ def column_sums_by_group(
             self.ptr, ind.ctypes.data, output.ctypes.data, num_threads
         )
         return output.T, lev
+
+
+@is_sparse.register
+def is_sparse_tatami(x: TatamiNumericPointer):
+    return x.sparse()
+
+
+@extract_dense_array.register
+def extract_dense_array_tatami(x: TatamiNumericPointer, subset: Tuple[Sequence[int], ...]) -> TatamiNumericPointer:
+    """See :py:meth:`~delayedarray.extract_dense_array.extract_dense_array`."""
+    return x.__generic_extract__(subset=subset)
+
+
+@extract_sparse_array.register
+def extract_sparse_array_tatami(x: TatamiNumericPointer, subset: Tuple[Sequence[int], ...]) -> TatamiNumericPointer:
+    """See :py:meth:`~delayedarray.extract_sparse_array.extract_sparse_array`."""
+    return x.__generic_extract__(subset=subset)
```

Comment:
This needs to return a `SparseNdarray`, which needs to be created on the C++ side; leave that to me.